apps/play/pages/games/cryptoUnicorns.tsx

```diff
@@ -12,85 +12,857 @@ import {
+      localhost: -1,
+    },
+  },
+  founders: {
+    poolIdByChain: {
+      mumbai: -1,
+      polygon: 41,
+      ethereum: -1,
+      localhost: -1,
+    },
+  },
+  communityCouncil: {
+    poolIdByChain: {
+      mumbai: -1,
+      polygon: 53,
+      ethereum: -1,
+      localhost: -1,
+    },
+  },
+  summerOfLoveTier1: {
+    poolIdByChain: {
+      mumbai: -1,
+      polygon: 67,
+      ethereum: -1,
+      localhost: -1,
+    },
+  },
+  summerOfLoveTier2: {
+    poolIdByChain: {
+      mumbai: -1,
+      polygon: 68,
+      ethereum: -1,
+      localhost: -1,
+    },
+  },
+  summerOfLoveTier3: {
+    poolIdByChain: {
+      mumbai: -1,
+      polygon: 69,
+      ethereum: -1,
+      localhost: -1,
+    },
+  },
+};
+
+const defaultLootboxBalances: { [key in terminusType]: number } = {
+  commonLootbox: 0,
+  rareLootbox: 0,
+  mythicLootbox: 0,
+  landLootbox: 0,
+  mysteryLootbox: 0,
+  RMPLootbox: 0,
+  goldenTicket: 0,
+  mythicKeystone: 0,
+  lightKeystone: 0,
+  wonderKeystone: 0,
+  mysteryKeystone: 0,
+  heartKeystone: 0,
+  cloudKeystone: 0,
+  flowerKeystone: 0,
+  candyKeystone: 0,
+  crystalKeystone: 0,
+  moonKeystone: 0,
+  rainbowKeystone: 0,
+  omnomKeystone: 0,
+  starKeystone: 0,
+  commonShadowcornEgg: 0,
+  rareShadowcornEgg: 0,
+  mythicShadowcornEgg: 0,
+  shadowcornTierOne: 0,
+  shadowcornTierTwo: 0,
+  shadowcornTierThree: 0,
+  nurseryTierOne: 0,
+  nurseryTierTwo: 0,
+  nurseryTierThree: 0,
+  iSurvivedLaunch: 0,
+  founders: 0,
+  communityCouncil: 0,
+  summerOfLoveTier1: 0,
+  summerOfLoveTier2: 0,
+  summerOfLoveTier3: 0,
+};
+
+interface NFTMetadata {
+  image: string;
+}
+
+// TODO(kellan): Add "locked" key. Reading this data will require us to extend the ERC721Metadata ABI to include:
+// 1. getUnicornMetadata for the Unicorns contract
+// 2. getLandMetadata for the Lands contract
+// 3. We need to check if Shadowcorns support game lock in their metadata
+interface NFTInfo {
+  tokenID: string;
+  tokenURI: string;
+  imageURI: string;
+  metadata: NFTMetadata;
+}
+
 const CryptoUnicorns = () => {
+  const [currentAccount, setCurrentAccount] = React.useState(
+    "0x0000000000000000000000000000000000000000"
+  );
   const [notEnoughRBW, setNotEnoughRBW] = React.useState(false);
   const [notEnoughUNIM, setNotEnoughUNIM] = React.useState(false);
   const [needAllowanceRBW, setNeedAllowanceRBW] = React.useState(false);
   const [needAllowanceUNIM, setNeedAllowanceUNIM] = React.useState(false);
   const [rbwToStash, setRBWToStash] = React.useState("");
   const [unimToStash, setUNIMToStash] = React.useState("");
+  const [terminusAddress, setTerminusAddress] = React.useState("");
+  const [UNIMAddress, setUNIMAddress] = React.useState("");
+  const [RBWAddress, setRBWAddress] = React.useState("");
+  const [unicornsAddress, setUnicornsAddress] = React.useState("");
+  const [landsAddress, setLandsAddress] = React.useState("");
+
+  const [lootboxBalances, setLootboxBalances] = React.useState({
```

Comment:
Rename this from `lootboxBalances` to `terminusBalances`. Doesn't have to happen on this PR. Refactor later.